server/aesdsocket.c

```diff
@@ -0,0 +1,363 @@
+/* Includes */
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <errno.h>
+#include <syslog.h>
+#include <netdb.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <linux/fs.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+
+
+/* Macros */
+
+#define BACKLOG_VAL 20 //Setting the maximum connection request limit while listening
+
+/* Global variables */
+int sock_fd = -1;
+int acc_fd = -1;
+int filetotest_fd = -1;
+char *packet_buffer;
+
+void sig_handler(int signum)
+{
+    if(signum == SIGINT)
+    {
+        syslog(LOG_DEBUG, "SIGINT received, exiting...");
+    }
+
+    else if(signum == SIGTERM)
+    {
+        syslog(LOG_DEBUG, "SIGTERM received, exiting...");
+    }
+
+    free(packet_buffer);
+
+    int status = close(sock_fd);
+    if(status < 0)
+    {
+        syslog(LOG_ERR, "Unable to close socket FD with an error: %d", errno);
+    }
+
+    status = close(acc_fd);
+    if(status < 0)
+    {
+        syslog(LOG_ERR, "Unable to close socket FD with an error: %d", errno);
+    }
+
+    status = close(filetotest_fd);
+    if(status < 0)
+    {
+        syslog(LOG_ERR, "Unable to close socket FD with an error: %d", errno);
+    }
+
+    unlink("/var/tmp/aesdsocketdata");
+    if(status < 0)
+    {
+        syslog(LOG_ERR, "Unable to close socket FD with an error: %d", errno);
+    }
+
+    closelog();
+    exit(EXIT_SUCCESS);
```

Comment:
@suja8742 are `free` and `exit` safe to do in a signal handler?

Reply:
My bad Dan, handled that in my current assignment.